src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs

```diff
@@ -2051,24 +2061,8 @@ private object GetSqlValueFromSqlBufferInternal(SqlBuffer data, _SqlMetaData met
             }
             else if (_typeSystem != SqlConnectionString.TypeSystem.SQLServer2000)
             {
-                // TypeSystem.SQLServer2005
```

Comment:
You have removed the check from `GetSqlValueFromSqlBufferInternal` and moved them to the callers. Why is that? 
All the callers of `GetSqlValueFromSqlBufferInternal` are throwing an exception on Udt.